agents/appmotion_motor/sat1_motors_driver.py

```diff
@@ -0,0 +1,825 @@
+                                        'pos': 0, #Position in counts
+                                        'realPos': 0.0, #Position in inches
+                                        'sPRev': 20000.0, #Steps per revolution (thread)
+                                        'isLin': motor2_isLin
+                                }
+
+                for motor in([self.motor1, self.motor2]):
+                        if motor:
+                                # Check to make sure the device is in receive mode and reset if necessary
+                                msg = motor.writeread(b'RS\r') #RS = Request Status
+                                motor.flushInput()
+                                print(msg)
+                                if (msg == b'RS=R'):
+                                        print("%s in receive mode." % (motor.propDict['name']))
+                                elif (msg != b'RS=R'):
+                                        print("%s not in receive mode.  Resetting." % (motor.propDict['name']))
+                                        print("Message was: ",msg)
+                                        self.killAllCommands(motor.propDict['motor'])
+                                        if (msg == b'RS=AR'):
+                                                amsg = motor.writeread(b'AL\r') #AL = Alarm Code
+                                                print('is message is: ',amsg)
+                                                print("Alarm was found. Resetting.")
+                                                motor.write(b'AR\r') #AR = Alarm Reset
+                                                motor.flushInput()
+                                        else:
+                                                print('Irregular message received.')
+                                                sys.exit(1)
+
+                                # Check the microstep resolution
+                                if mRes:
+                                        motor.write(b'EG8000\r') #EG = Electronic Gearing
+                                        motor.write(b'SA\r') #SA = Save Parameters
+                                        motor.flushInput()
+                                        sleep(0.1)
+                                        msg = motor.writeread(b'EG\r')
+                                        motor.flushInput()
+                                        if(len(msg) <= 4):    # Need at least MR=X + \r, which is 5 characters
+                                                print("Couldn't get microstep resolution for %s.  Assuming 8." % (motor.propDict['name']))
+                                        elseis:
+                                                print(msg)
+#                                                 msInfo = msg.rstrip('\r')[3:]
+                                                motor.propDict['sPRev'] = float(msInfo)
```

Comment:
When we send the code `EG` to the motor controllers it returns a code like this `EG=number` and we only want the number. So that line is just slicing the returned code to grab the number. I also dropped the `rstrip` method so it now looks like `ms_info = msg[3:]`.